src/go/accessd/config/config.go

```diff
@@ -0,0 +1,165 @@
+// LICENSE file in the root directory of this source tree.
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/pkg/errors"
+	"google.golang.org/grpc/resolver"
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/proto"
+
+	"github.com/magma/magma/accessd/config/internal/grpcutil"
+	"github.com/magma/magma/log"
+	"github.com/magma/magma/protos/magma/mconfig"
+)
+
+// LogLevel translates protobuf defined mconfig.AccessD_LogLevel to log.Level.
+func LogLevel(l mconfig.AccessD_LogLevel) log.Level {
+	switch l {
+	case mconfig.AccessD_DEBUG:
+		return log.DebugLevel
+	case mconfig.AccessD_INFO:
+		return log.InfoLevel
+	case mconfig.AccessD_WARN:
+		return log.WarnLevel
+	case mconfig.AccessD_ERROR:
+		return log.ErrorLevel
+	}
+	return log.InfoLevel
+}
+
+const (
+	ipv4Scheme = "ipv4"
+	ipv6Scheme = "ipv6"
+)
+
+// ParseTarget takes a target in string form and returns a resolved Target.
+// Extends functionality in grpc/internal/grpcutil.ParseTarget to support ipv4
+// and ipv6 schemes.
+func ParseTarget(target string) resolver.Target {
+	if strings.HasPrefix(target, ipv4Scheme+":") {
+		return resolver.Target{
+			Scheme:   ipv4Scheme,
+			Endpoint: target[len(ipv4Scheme)+1:],
+		}
+	}
+	if strings.HasPrefix(target, ipv6Scheme+":") {
+		return resolver.Target{
+			Scheme:   ipv6Scheme,
+			Endpoint: target[len(ipv6Scheme)+1:],
+		}
+	}
+	return grpcutil.ParseTarget(target, false)
+}
+
+// Configer returns a parsed config.
+type Configer interface {
+	Config() *mconfig.AccessD
+}
+
+// ConfigManager implements Configer via a loaded config.
+type ConfigManager struct {
+	config *mconfig.AccessD
+}
+
+var defaultConfig = &mconfig.AccessD{
+	LogLevel:                        mconfig.AccessD_INFO,
+	SctpdDownstreamServiceTarget:    "unix:///tmp/sctpd_downstream.sock",
+	SctpdUpstreamServiceTarget:      "unix:///tmp/sctpd_upstream.sock",
+	MmeSctpdDownstreamServiceTarget: "unix:///tmp/mme_sctpd_downstream.sock",
+	MmeSctpdUpstreamServiceTarget:   "unix:///tmp/mme_sctpd_upstream.sock",
+}
+
+const (
+	cStyleCommentStart = "/*"
+	cStyleCommentEnd   = "*/"
+)
+
+func filterCStyleComments(in string) string {
+	var filtered string
+	for {
+		idx := strings.Index(in, cStyleCommentStart)
+		if idx == -1 {
+			filtered += in
+			break
+		}
+		filtered += in[:idx]
+		in = in[idx:]
+		idx = strings.Index(in, cStyleCommentEnd)
+		if idx == -1 {
+			break
+		}
+		in = in[idx+2:]
+	}
+	return filtered
+}
+
+func filterJSONComments(in string) string {
+	var filtered []string
+	for _, line := range strings.Split(filterCStyleComments(in), "\n") {
```

Comment:
We want to support comments in JSON files for documentation purposes. Other popular progarms (e.g. VSCode) use a settings.json file and support comments.

(Also, a stated reason behind wanting to use YAML as a config language is because it supports comments. With this addition, we are one step closer to eliminating YAML :) )

Background:

1. JSON spec does not support comments (they are explicitly disallowed, https://en.wikipedia.org/wiki/JSON#cite_note-DouglasCrockfordComments-24). A pre-processor is recommended/suggested
2. Golang's built in json parsing does not support comments. This is by design/follows their conventions: they are very strict about adherence to official specs.
3. AFAICT, no well tested pre-processor exists. I found a few, but they don't seem to have good tests, e.g.

https://github.com/wuxiangzhou2010/jsonuncommenter/blob/master/uncommenter_test.go
https://github.com/muhammadmuzzammil1998/jsonc/blob/master/jsonc_test.go

4. Overall, this pre-processor is ~30 lines of code easily testable. Not quite a "leftpad" situation, but pretty close :)